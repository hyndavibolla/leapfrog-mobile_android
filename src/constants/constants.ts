export const pushNotificationTypes = {
  USER_INACTIVE: 'USER_INACTIVE',
  MISSION_COMPLETE: 'MISSION_COMPLETE',
  BALANCE_NEW_POINTS: 'BALANCE_NEW_POINTS',
  BALANCE_EXPIRING_POINTS: 'BALANCE_EXPIRING_POINTS',
  BALANCE_UNUSED_POINTS: 'BALANCE_UNUSED_POINTS',
  MARKETING: 'MARKETING',
  USER_BIRTHDAY: 'USER_BIRTHDAY'
};

export const RedemptionMethods = {
  IN_STORE_ONLY: 'In-Store only',
  ONLINE_ONLY: 'Online only',
  IN_STORE_OR_ONLINE_ONLY: 'In-Store and Online only',
  IN_RESTAURANT_ONLY: 'In Restaurant only',
  APP_ONLY: 'App only'
};

export const GROCERY_AND_DELIVERY_CATEGORY = 'Grocery & Delivery';

export const TutorialFrom = {
  TUTORIAL_REMINDER: 'tutorial_reminder',
  TUTORIAL_BANNER: 'tutorial_banner',
  PROFILE: 'profile'
};
