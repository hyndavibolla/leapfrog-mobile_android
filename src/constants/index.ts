export * from './env';
export * from './routes';
export * from './eventTracking';
export * from './styles';
export * from './icons';
export * from './constants';
