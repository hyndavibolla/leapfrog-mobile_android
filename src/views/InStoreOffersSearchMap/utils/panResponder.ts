const swipeConfig = {
  velocityThreshold: 0.3,
  directionalOffsetThreshold: 80,
  gestureIsClickThreshold: 5
};

const swipeDirections = {
  swipeUp: 'SWIPE_UP',
  swipeDown: 'SWIPE_DOWN'
};

export const handleShouldSetPanResponder = ({ nativeEvent: { touches } }, gestureState) => {
  return touches.length === 1 && !gestureIsClick(gestureState);
};

export const gestureIsClick = ({ dx, dy }) => {
  return Math.abs(dx) < swipeConfig.gestureIsClickThreshold && Math.abs(dy) < swipeConfig.gestureIsClickThreshold;
};

export const handlePanResponderEnd = (gestureState, onSwipeUp, onSwipeDown) => {
  const swipeDirection = getSwipeDirection(gestureState);
  triggerSwipeHandlers(swipeDirection, onSwipeUp, onSwipeDown);
};

export const triggerSwipeHandlers = (swipeDirection, onSwipeUp, onSwipeDown) => {
  const { swipeUp, swipeDown } = swipeDirections;
  switch (swipeDirection) {
    case swipeUp:
      onSwipeUp();
      break;
    case swipeDown:
      onSwipeDown();
      break;
  }
};

export const getSwipeDirection = gestureState => {
  const { swipeUp, swipeDown } = swipeDirections;
  if (isValidVerticalSwipe(gestureState)) {
    return gestureState.dy > 0 ? swipeDown : swipeUp;
  }
  return null;
};

export const isValidVerticalSwipe = ({ vy, dx }) => {
  const { velocityThreshold, directionalOffsetThreshold } = swipeConfig;
  return isValidSwipe(vy, velocityThreshold, dx, directionalOffsetThreshold);
};

export const isValidSwipe = (velocity, velocityThreshold, directionalOffset, directionalOffsetThreshold) => {
  return Math.abs(velocity) > velocityThreshold && Math.abs(directionalOffset) < directionalOffsetThreshold;
};
