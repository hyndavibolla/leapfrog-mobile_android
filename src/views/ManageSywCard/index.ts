export { default as ManageSywCard } from './ManageSywCard';
