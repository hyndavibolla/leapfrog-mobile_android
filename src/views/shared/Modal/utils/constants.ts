export enum ModalType {
  CENTER = 'center',
  BOTTOM = 'bottom'
}

export enum ModalSize {
  SMALL = 'small',
  MEDIUM = 'medium',
  EXTRA_LARGE = 'extralarge',
  DYNAMIC = 'dynamic',
  FULL_SCREEN = 'fullscreen'
}

export const ModalSizeToHeightMap = {
  [ModalSize.SMALL]: 40,
  [ModalSize.MEDIUM]: 50,
  [ModalSize.EXTRA_LARGE]: 80,
  [ModalSize.FULL_SCREEN]: 100
};
