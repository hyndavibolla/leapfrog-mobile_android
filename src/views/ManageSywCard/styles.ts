import { css } from '@emotion/native';

import { COLOR, FONT_FAMILY, FONT_SIZE } from '../../constants';

export const styles = {
  contentContainer: css`
    flex: 1;
    justify-content: center;
  `,
  imageContainer: css`
    margin: 0 32px 16px;
  `,
  image: css`
    align-self: center;
    border-radius: 4px;
  `,
  sectionContainer: css`
    margin: 0 16px;
  `,
  title: css`
    font-family: ${FONT_FAMILY.HEAVY};
    color: ${COLOR.BLACK};
    text-align: center;
    margin-top: 20px;
    margin-bottom: 8px;
  `,
  text: css`
    text-align: center;
    font-size: ${FONT_SIZE.SMALLER};
    line-height: ${FONT_SIZE.HUGE};
    color: ${COLOR.DARK_GRAY};
  `,
  textSpacing: css`
    margin: 8px 0;
  `,
  link: css`
    color: ${COLOR.PRIMARY_BLUE};
    letter-spacing: 0;
  `,
  buttonContainer: css`
    margin: 8px 40px 20px;
  `,
  buttonOuter: css`
    width: 100%;
  `,
  buttonInner: css`
    padding: 8px 0;
    background-color: ${COLOR.PRIMARY_BLUE};
  `,
  buttonText: css`
    font-size: ${FONT_SIZE.SMALL};
    letter-spacing: 0;
    line-height: ${FONT_SIZE.HUGE};
    font-family: ${FONT_FAMILY.BOLD};
  `,
  contactText: css`
    color: ${COLOR.DARK_GRAY};
    font-size: ${FONT_SIZE.SMALLER};
    line-height: ${FONT_SIZE.HUGE};
    font-size: ${FONT_SIZE.TINY};
    text-align: center;
    text-transform: uppercase;
    letter-spacing: 0;
  `
};
