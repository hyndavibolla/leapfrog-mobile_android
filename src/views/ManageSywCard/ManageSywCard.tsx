import React, { memo, useContext, useMemo } from 'react';
import { View, Image, ScrollView } from 'react-native';

import { GlobalContext } from '../../state-mgmt/GlobalState';
import { Title, TitleType } from '../shared/Title/Title';
import { Button } from '../shared/Button';
import { useTestingHelper } from '../../utils/useTestingHelper';
import { ENV, FONT_FAMILY } from '../../constants';
import { Text } from '../shared/Text';

import { styles } from './styles';

export const ManageSywCard = () => {
  const { deps } = useContext(GlobalContext);
  const { getTestIdProps } = useTestingHelper('manage-syw-card');
  const ecmWidth = useMemo(() => deps.nativeHelperService.dimensions.getWindowWidth() - 60, [deps.nativeHelperService.dimensions]);
  const ecmHeight = useMemo(() => ecmWidth / 1.4, [ecmWidth]);

  return (
    <ScrollView contentContainerStyle={styles.contentContainer} {...getTestIdProps('scroll-view')}>
      <View style={styles.imageContainer}>
        <Image source={require('../../assets/credit-card/creditCard.png')} style={styles.image as any} />
      </View>
      <View style={[styles.sectionContainer, { paddingBottom: ecmHeight / 10 }]}>
        <Title type={TitleType.HEADER} style={styles.title}>
          Manage your card {'\n'} from Citibank
        </Title>
        <Text style={[styles.text, styles.textSpacing]}>View your transactions, pay your bills, check your expenses and more.</Text>
        <View style={styles.buttonContainer}>
          <Button
            innerContainerStyle={[styles.buttonOuter, styles.buttonInner]}
            textStyle={styles.buttonText}
            onPress={() => deps.nativeHelperService.linking.openURL(ENV.CITI.LANDING_URL)}
            {...getTestIdProps('manage-card-button')}
          >
            Manage Card
          </Button>
        </View>
        <Text font={FONT_FAMILY.BOLD} style={[styles.text, styles.contactText]}>
          or contact:
        </Text>
        <Text
          font={FONT_FAMILY.BOLD}
          style={[styles.text, styles.link]}
          onPress={() => deps.nativeHelperService.linking.openURL(`tel:${ENV.CITI.PHONE_NUMBER}`)}
          {...getTestIdProps('phone-link')}
        >
          {ENV.CITI.PHONE_NUMBER}
        </Text>
        <Text style={[styles.text, styles.contactText]}>Available 24/7</Text>
      </View>
    </ScrollView>
  );
};

export default memo(ManageSywCard);
