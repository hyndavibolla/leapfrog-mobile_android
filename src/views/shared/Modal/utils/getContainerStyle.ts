import { memorize } from '../../../../utils/memorize';
import { styles } from '../styles';
import { ModalSize, ModalType, ModalSizeToHeightMap } from './constants';

interface props {
  height: number;
  modalHeight: number;
  size: ModalSize;
  type: ModalType;
}

const getContainerStyle = ({ height, modalHeight, size, type }: props) => {
  switch (type) {
    case ModalType.BOTTOM:
      if (size === ModalSize.FULL_SCREEN) return [styles.containerBottom, styles.containerBottomFullScreen];
      else if (size !== ModalSize.DYNAMIC) return [styles.containerBottom, { maxHeight: (ModalSizeToHeightMap[size] * height) / 100 }];
      return [styles.containerBottom];
    case ModalType.CENTER:
      if (size === ModalSize.DYNAMIC) return [styles.containerCenter, styles.containerCenterDynamic, { bottom: (height - modalHeight) / 2 }];
      const center = (ModalSizeToHeightMap[size] * height) / 100;
      return [styles.containerCenter, { height: center, bottom: (height - center) / 2 }];
  }
};

export default memorize(getContainerStyle);
