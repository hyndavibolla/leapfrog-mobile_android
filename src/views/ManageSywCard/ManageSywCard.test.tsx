import React from 'react';
import { act } from 'react-test-renderer';
import { fireEvent } from '@testing-library/react-native';

import { renderWithGlobalContext } from '../../test-utils/renderWithGlobalContext';
import { ManageSywCard } from './ManageSywCard';
import { wait } from '../../utils/wait';
import { ENV } from '../../constants';
import { getMockDeps } from '../../test-utils/getMockDeps';
import { Deps } from '../../models/general';

describe('Apply Now', () => {
  let deps: Deps;

  beforeEach(() => {
    deps = getMockDeps();
  });

  it('should render', async () => {
    const { toJSON, queryByTestId } = renderWithGlobalContext(<ManageSywCard />);
    await act(() => wait(0));
    expect(toJSON()).toMatchSnapshot();
    expect(queryByTestId('manage-syw-card-scroll-view')).toBeTruthy();
  });

  it('should call citi by phone', async () => {
    const { queryByTestId } = renderWithGlobalContext(<ManageSywCard />, deps);
    await act(() => wait(0));
    fireEvent.press(await queryByTestId('manage-syw-card-phone-link'));
    expect(deps.nativeHelperService.linking.openURL).toBeCalledWith(expect.stringContaining(ENV.CITI.PHONE_NUMBER));
  });

  it('should call citi landing', async () => {
    const { queryByTestId } = renderWithGlobalContext(<ManageSywCard />, deps);
    await act(() => wait(0));
    fireEvent.press(await queryByTestId('manage-syw-card-manage-card-button'));
    expect(deps.nativeHelperService.linking.openURL).toBeCalledWith(ENV.CITI.LANDING_URL);
  });
});
