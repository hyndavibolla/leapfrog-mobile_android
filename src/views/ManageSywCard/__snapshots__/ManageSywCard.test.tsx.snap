// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apply Now should render 1`] = `
<RCTScrollView
  accessibilityLabel="manage-syw-card-scroll-view"
  accessible={true}
  contentContainerStyle={
    Object {
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
    }
  }
  testID="manage-syw-card-scroll-view"
>
  <View>
    <View
      style={
        Object {
          "marginBottom": 16,
          "marginLeft": 32,
          "marginRight": 32,
          "marginTop": 0,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/credit-card/creditCard.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "borderBottomLeftRadius": 4,
            "borderBottomRightRadius": 4,
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 0,
            "marginLeft": 16,
            "marginRight": 16,
            "marginTop": 0,
          },
          Object {
            "paddingBottom": 49.28571428571429,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#262C2F",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": 20,
              "lineHeight": 30,
            },
            Object {
              "color": "#262C2F",
              "fontFamily": "SFProDisplay-Heavy",
              "marginBottom": 8,
              "marginTop": 20,
              "textAlign": "center",
            },
          ]
        }
        type="titleHeader"
      >
        Manage your card 
        

         from Citibank
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#909BA0",
              "fontFamily": "SFProDisplay-Medium",
              "fontSize": 16,
              "lineHeight": 22,
            },
            Array [
              Object {
                "color": "#909BA0",
                "fontSize": 16,
                "lineHeight": 26,
                "textAlign": "center",
              },
              Object {
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 8,
              },
            ],
          ]
        }
      >
        View your transactions, pay your bills, check your expenses and more.
      </Text>
      <View
        style={
          Object {
            "marginBottom": 20,
            "marginLeft": 40,
            "marginRight": 40,
            "marginTop": 8,
          }
        }
      >
        <View
          accessibilityLabel="manage-syw-card-manage-card-button"
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="manage-syw-card-manage-card-button"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#0066CC",
                  "borderBottomLeftRadius": 30,
                  "borderBottomRightRadius": 30,
                  "borderTopLeftRadius": 30,
                  "borderTopRightRadius": 30,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 8,
                },
                Array [
                  Object {
                    "width": "100%",
                  },
                  Object {
                    "backgroundColor": "#0066CC",
                    "paddingBottom": 8,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 8,
                  },
                ],
              ]
            }
          >
            <View
              style={false}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#909BA0",
                      "fontFamily": "SFProDisplay-Medium",
                      "fontSize": 16,
                      "lineHeight": 22,
                    },
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 14,
                        "textAlign": "center",
                      },
                      Object {
                        "fontFamily": "SFProDisplay-Bold",
                        "fontSize": 18,
                        "letterSpacing": 0,
                        "lineHeight": 26,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Manage Card
              </Text>
            </View>
          </View>
        </View>
      </View>
      <Text
        font="SFProDisplay-Bold"
        style={
          Array [
            Object {
              "color": "#909BA0",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": 16,
              "lineHeight": 22,
            },
            Array [
              Object {
                "color": "#909BA0",
                "fontSize": 16,
                "lineHeight": 26,
                "textAlign": "center",
              },
              Object {
                "color": "#909BA0",
                "fontSize": 12,
                "letterSpacing": 0,
                "lineHeight": 26,
                "textAlign": "center",
                "textTransform": "uppercase",
              },
            ],
          ]
        }
      >
        or contact:
      </Text>
      <Text
        accessibilityLabel="manage-syw-card-phone-link"
        accessible={true}
        font="SFProDisplay-Bold"
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#909BA0",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": 16,
              "lineHeight": 22,
            },
            Array [
              Object {
                "color": "#909BA0",
                "fontSize": 16,
                "lineHeight": 26,
                "textAlign": "center",
              },
              Object {
                "color": "#0066CC",
                "letterSpacing": 0,
              },
            ],
          ]
        }
        testID="manage-syw-card-phone-link"
      >
        1-877-816-9063
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#909BA0",
              "fontFamily": "SFProDisplay-Medium",
              "fontSize": 16,
              "lineHeight": 22,
            },
            Array [
              Object {
                "color": "#909BA0",
                "fontSize": 16,
                "lineHeight": 26,
                "textAlign": "center",
              },
              Object {
                "color": "#909BA0",
                "fontSize": 12,
                "letterSpacing": 0,
                "lineHeight": 26,
                "textAlign": "center",
                "textTransform": "uppercase",
              },
            ],
          ]
        }
      >
        Available 24/7
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
